src/protocol/receiver/time.rs

```diff
@@ -0,0 +1,111 @@
+use std::time::{Duration, Instant};
+
+use stats::OnlineStats;
+
+use crate::protocol::{TimeBase, TimeSpan, TimeStamp};
+
+pub(crate) struct SynchronizedRemoteClock {
+    tolerance: Duration,
+    time_base: TimeBase,
+    stats: Option<OnlineStats>,
+}
+
+impl SynchronizedRemoteClock {
+    const MAX_SAMPLES: usize = 1_000;
+    const DRIFT_TOLERANCE: Duration = Duration::from_millis(5);
+
+    pub fn new(now: Instant) -> Self {
+        Self {
+            // TODO: Drift deviation tolerance should be parameterized.
+            //       It wasn't in the reference implementation, but I added it because the reference
+            //       implementation is susceptible to invalid clock adjustments during periods of
+            //       acute network latency
+            tolerance: Self::DRIFT_TOLERANCE,
+            time_base: TimeBase::new(now),
+            stats: None,
+        }
+    }
+
+    pub fn synchronize(&mut self, now: Instant, ts: TimeStamp) {
+        let drift = self.time_base.timestamp_from(now) - ts;
```

Comment:
Isn't it incorrect that these should be the same, shouldn't now be `0.5 * rtt` after `ts`?